vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSlideController.java

```diff
@@ -10,27 +10,39 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 
+import edu.asu.diging.vspace.core.model.IModule;
 import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.impl.ModuleManager;
 import edu.asu.diging.vspace.web.staff.forms.SlideForm;
 
 @Controller
 public class AddSlideController {
 
     @Autowired
     private ISlideManager slideManager;
+    
+    @Autowired
+    private ModuleManager moduleManager;
 
     @RequestMapping(value = "/staff/module/{id}/slide/add", method = RequestMethod.GET)
     public String showAddSlide(@PathVariable("id") String moduleId, Model model) {
         model.addAttribute("moduleId", moduleId);
         model.addAttribute("slide", new SlideForm());
+        model.addAttribute("sequences", moduleManager.getModuleSequences(moduleId));
+        
         return "staff/module/slide/add";
     }
 
     @RequestMapping(value = "/staff/module/{moduleId}/slide/add", method = RequestMethod.POST)
     public String addSlide(Model model, @PathVariable("moduleId") String moduleId, @ModelAttribute SlideForm slideForm,
             Principal principal) {
 
-        slideManager.createSlide(moduleId, slideForm);
+        IModule module = moduleManager.getModule(moduleId);
+        if(slideForm.getSlideType().substring(0, slideForm.getSlideType().length()-1).equals("slide")) {
```

Comment:
why this? what does slide type contain?

Reply:
also, do not hardcode strings!

Reply:
And third, I would always use the most basic type in the else block. If someone submits "chicken" as type you want a slide to be created (if anything).